packages/cli/test/functional/test.js

```diff
@@ -0,0 +1,69 @@
+const path = require('path');
+const fs = require('fs-extra-promise');
+const { execSync } = require('child_process');
+
+const { compare } = require('./testUtil/compare');
+const { cleanupConvert } = require('./testUtil/cleanup');
+
+const {
+  testSites,
+  testConvertSites,
+  testTemplateSites,
+} = require('./testSites');
+
+/* eslint-disable no-console */
+
+function printFailedMessage(err, siteName) {
+  console.log(err);
+  console.log(`Test result: ${siteName} FAILED`);
+}
+
+const execOptions = {
+  stdio: ['inherit', 'inherit', 'inherit'],
+};
+
+testSites.forEach((siteName) => {
+  console.log(`Running ${siteName} tests`);
+  try {
+    execSync(`node ../../index.js build ${siteName}`, execOptions);
+    compare(siteName);
+  } catch (err) {
+    printFailedMessage(err, siteName);
+    process.exit(1);
+  }
+});
+
+testConvertSites.forEach((siteName) => {
+  console.log(`Running ${siteName} tests`);
```

Comment:
I feel like we can further break these into separate `testXYZSite` functions.

Reply:
```
(siteName) => {
  console.log(`Running ${siteName} tests`);
  try {
    execSync(`node ../../index.js build ${siteName}`, execOptions);
    compare(siteName);
  } catch (err) {
    printFailedMessage(err, siteName);
    process.exit(1);
  }
}
```

you mean to reuse this for all three test types? Sounds good!

Reply:
Hmm I meant having separate `testConvertSite`, `testTemplateSite` and `testSite` functions, with each function doing what is currently done in the anonymous function we currently have. Not sure if this is what you meant!

Reply:
Hmm I meant something like this:
```
genericTestFunction = (siteName, cleanup) => {
  console.log(`Running ${siteName} tests`);
  try {
    execSync(`node ../../index.js build ${siteName}`, execOptions);
    compare(siteName);
  } catch (err) {
    printFailedMessage(err, siteName);
    process.exit(1);
    cleanup()
  }
  cleanup()
}

testConvertSites.forEach((siteName) => {
  genericTestFunction(siteName, cleanupCallback, ...)
})
```

Did you mean to just extract the anonymous functions as named functions?